File: lib/ios/MapView/HeatMapManager.m
#import <React/RCTUIManager.h>

@interface RCT_EXTERN_MODULE(AMapHeatMapManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(data, NSArray)
RCT_EXPORT_VIEW_PROPERTY(radius, int)
RCT_EXPORT_VIEW_PROPERTY(opacity, double)

@end

File: lib/ios/MapView/HeatMapManager.swift
@objc(AMapHeatMapManager)
class AMapHeatMapManager: RCTViewManager {
  override class func requiresMainQueueSetup() -> Bool {
    false
  }

  override func view() -> UIView {
    HeatMap()
  }
}

class HeatMap: UIView, Overlay {
  var overlay: MABaseOverlay? = nil
  var renderer: MAHeatMapVectorOverlayRender?

  func getRenderer() -> MAOverlayRenderer {
    if (renderer == nil) {
      renderer = MAHeatMapVectorOverlayRender(heat: (overlay as! MAHeatMapVectorOverlay))
    }
    return renderer!
  }

  @objc func setData(_ data: NSArray) {
  }

  @objc func setRadius(_ radius: Double) {
  }

  @objc func setOpacity(_ opacity: Double) {
  }
}